crates/storage/provider/src/bundle_state/bundle_state_with_receipts.rs

```diff
@@ -241,20 +242,22 @@ impl BundleStateWithReceipts {
     /// of receipt. This is a expensive operation.
     pub fn receipts_root_slow(&self, block_number: BlockNumber) -> Option<H256> {
         let index = self.block_number_to_index(block_number)?;
-        let block_receipts =
-            self.receipts[index].iter().map(Option::as_ref).collect::<Option<Vec<_>>>()?;
+        let block_receipts = self.receipts.receipt_vec[index]
+            .iter()
+            .map(Option::as_ref)
+            .collect::<Option<Vec<_>>>()?;
```

Comment:
It would be good to move this inside `Receipts` and call `self.receipts.root_slow(index)`

Reply:
@rakita Done